doc/sphinx/source/android.rst

```diff
@@ -0,0 +1,79 @@
+
+It is recommended to Build PoCL as an external project in the CMakeLists.txt that belongs to the native code of the
+Android project. This will set the ``ANDROID_NDK, ANDROID_PLATFORM_LEVEL`` and ``ANDROID_ABI`` to what you are building the
+project for. By default, Android Studio will build native code for multiple architectures (ARM 32/64 and x86), so the
+``ANDROID_ABI`` will change for each architecture. Adding pocl as a library dependency to your native code will ensure that
+it is packed into the APK. It is recommended to set ``-DBUILD_SHARED_LIBS=0`` so that PoCL gets built as a static library
+(libpocl.a) as this is easier to use.
+
+Building Remote Client
+----------------------
+
+If you want to make use of PoCL-R, you can add ``-DENABLE_REMOTE_CLIENT=YES`` to the cmake options
+and make sure that network access is allowed in the `AndroidManifest.xml`.
+
+
+Building Proxy Device
+---------------------
+
+To make use of the Proxy device on Android, You first need to make sure that the phone comes with an OpenCL library and
+that is whitelisted by the vendor. Starting with API level 24, vendors need whitelist libraries that are allowed to be dlopened.
+To check that OpenCL is whitelisted do this:
+
+    1. adb into the phone
+    2. run::
+
+        cat /vendor/etc/public.libraries.txt
+
+    3. check that `libOpenCL.so` is there
+
+For newer Android versions (Android 12 and up), you also need to add::
+
+        <uses-native-library
+            android:name="libOpenCL.so"
+            android:required="false" />
+
+to the ``<applications>`` element of the `AndroidManifest.xml`
+
+Once you know that you the phone comes with an OpenCL library, it possible to use the proxy device. To build the proxy device add the
+following CMake options to the ones mentioned before: ``-DENABLE_PROXY_DEVICE=YES -DVISIBILITY_HIDDEN=NO``. This will build
+the proxy device and pocl as a static library. If you want to use JOCL, you need to also add ``-DPROXY_USE_LIBOPENCL_STUB=YES``
+and set ``-DBUILD_SHARED_LIBS=YES``. This will build a dynamic library of pocl.
+
+*NOTE:* The proxy driver suffers from the same issues the remote driver has with :ref:`Mali GPUs<remote-issues-label>`.
+See that section for a workaround.
+
+
+Setting PoCL Environment Variables
+----------------------------------
+
+The easiest way to set PoCL environment variables is to create a native function that calls stdlib.h's setenv function
+
+Using JOCL
+----------
+
+It is possible to use JOCL on Android. However, by default JOCL does not get built for Android. It also doesn't look for libpocl.
```

Comment:
We shall upstream these changes to JOCL, perhaps point to the JOCL branch which are being upstreamed (PR) directly?

Reply:
I've made a pr and will change the docs to reflect this if it gets accepted